src/supported-operations/thin-edge-supported-operations.md

```diff
@@ -0,0 +1,118 @@
+3. Mapper takes the filename as operation name and reads the operation file.
+4. Empty operation file means there are no additional configuration parameters and mapper is ready to send supported operations message to c8y which contains the list of supported operations i.e. `c8y_Restart`.
+5. Cloud operator wants restart the device therefore they send the operation message to the device which mapper interprets as a restart operation and mapper executes restart.
+
+## Example 2 (special case)
+
+Given operation `c8y_LogfileRequest`, the operation file would be created in `/etc/tegde/operations/c8y` with the filename `c8y_LogfileRequest` so the full path would be `/etc/tegde/operations/c8y/c8y_LogfileRequest`.
+Operation `c8y_LogfileRequest` requires additional parameter `log_type` to be set.
+
+With the operation file the following structure is written:
+
+```toml
+[exec]
+
+command = "/etc/tedge/plugins/c8y_LogfileRequest"
+user = "root"
+
+[init]
+topic = "c8y/s/us"
+message = "116,error"
+
+[extras]
+  # Additional configuration if the operation requires additional configuration
+  log_type = ["error"]
+```
+
+> Alternatives:
+>
+> Option 1: on init of mapper config will contain a command to be called e.g. sent `log_type` message to c8y, only the component knows about it (mapper doesn't need to care about it)
+> Option 2: explicit additional message to be send by the mapper if required/defined using a table in config file e.g. `[init]` like in the example above.
+>
+> Done
+
+Given a mapper the flow would be as follows:
+
+1. Mapper reads directory for the cloud which it supports i.e. `/etc/tegde/operations/c8y`.
+2. Mapper finds the operation file for the operation `c8y_LogfileRequest` in the directory `c8y` with filename `c8y_LogfileRequest` which is the only operation.
+3. Mapper takes the filename as operation name and reads the operation file.
+4. Operation file contains additional configuration parameters and mapper shall know (i.e. implements) how to interpret the configuration (in this case mapper has to add new fragment to the device type therefore can read the `log_type` configuration and send new `log_type` message) ready to send supported operations message to c8y which contains the list of supported operations i.e. `c8y_LogfileRequest`. E.g. `114,c8y_LogfileRequest`
+5. Cloud operator wants to retrieve logs from the device therefore they send the operation message to the device which mapper interprets as a `c8y_LogfileRequest` operation and mapper executes `c8y_LogfileRequest`.
+
+## Example 3
+
+Given an operation which requires communication with another component over the bus e.g. `c8y_SoftwareUpdate` and no additional configuration is required.
+
+Following operation file could be used:
+
+```toml
+[mqtt]
+
+request = "tedge/commands/req/software/update"
+response = "tedge/commands/res/software/update"
```

Comment:
With this scheme that allows thin-edge to communicate with the plugin over MQTT topics, where do we define the expected message format as these configurations only define the topics? Or we're going to pass the messages received from the cloud as-is to these topics? If that's that case I wonder what the point of this topic redirection would be.